### PR TITLE
scala fs2-grpc and zio-grpc implementation

### PR DESCRIPTION
**Pull request by jtjeferreira**
a bit better than scala akka but still some way to go until it reaches java performance

### CODE REVIEW

**Comment by jtjeferreira** [comment]
updated the PR with a zio-grpc implementation

**Comment by LesnyRumcajs** [comment]
@jtjeferreira Thanks! How about we grab the most performing implementation / executor setting / else for Scala and put the rest to https://github.com/LesnyRumcajs/grpc_bench/tree/master/detailed ?

**Comment by jtjeferreira** [comment]
> @jtjeferreira Thanks! How about we grab the most performing implementation / executor setting / else for Scala and put the rest to [https://github.com/LesnyRumcajs/grpc_bench/tree/master/detailed](https://github.com/LesnyRumcajs/grpc_bench/tree/master/detailed?rgh-link-date=2021-05-18T19%3A57%3A29Z) ?

Hi @LesnyRumcajs can we keep these 3, and I promise I wont add any new ones? These represent the main projects/organizations in the scala world...

Btw: After a few hours wasted of profiling I noticed the heap settings were not being applied correctly, so I hope the results will be better (but still not close to the java implementation)

File: scala_zio_bench/src/main/protobuf/helloworld.proto [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2015 gRPC authors.
```
Comment (LesnyRumcajs) [inline scala_zio_bench/src/main/protobuf/helloworld.proto:1]: let's copy this guy from proto/ in the dockerfile, like in e.g. Java implementations

Comment (LesnyRumcajs) [reply scala_zio_bench/src/main/protobuf/helloworld.proto:1]: In fact it seems you're already doing that, not sure why this file is added?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

**Comment by LesnyRumcajs** [comment]
> > @jtjeferreira Thanks! How about we grab the most performing implementation / executor setting / else for Scala and put the rest to [https://github.com/LesnyRumcajs/grpc_bench/tree/master/detailed](https://github.com/LesnyRumcajs/grpc_bench/tree/master/detailed?rgh-link-date=2021-05-18T19%3A57%3A29Z) ?
> 
> Hi @LesnyRumcajs can we keep these 3, and I promise I wont add any new ones? These represent the main projects/organizations in the scala world...
> 
> Btw: After a few hours wasted of profiling I noticed the heap settings were not being applied correctly, so I hope the results will be better (but still not close to the java implementation)

Sure, if you say those are completely distinct projects / frameworks then I see no problem with that.

File: scala_zio_bench/src/main/protobuf/helloworld.proto (hunk 1 above)
Comment (jtjeferreira) [reply scala_zio_bench/src/main/protobuf/helloworld.proto]: I added the file so I can develop locally (i.e build outside the docker container), but forgot to remove it...

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**